File: 2395-longest-binary-subsequence-less-than-or-equal-to-k/longest-binary-subsequence-less-than-or-equal-to-k.js
/**
 * @param {string} s
 * @param {number} k
 * @return {number}
 */
var longestSubsequence = function(s, k) {
    let num=0;
    let bits=Math.log2(k)+1;
    let count=0;
    let n =s.length-1;
    let j=0;
    while(n>=0){
        const ch =s[n];
        if(ch==="1"){
            if(j<bits && num+(1<<j)<=k){
                count++;
                num+=(1<<j);
                
            }
            
        }
        else{
            count++;
        }
j++;n--;
        
    }
    return count;
};

File: 2395-longest-binary-subsequence-less-than-or-equal-to-k/longest-binary-subsequence-less-than-or-equal-to-k.cpp
class Solution {
public:
    int longestSubsequence(string s, int k) {
        int curr=0;
        int count =0;
        int bits=floor(log2(k)) + 1; 
        for(int i =0;i<s.size();i++){
            char ch = s[s.size()-1-i];
            if(ch=='1'){
                if(i<bits && curr+(1<<i)<=k){
                    count++;
                    curr+=1<<i;
                }
            }
            else{
                count++;
            }
        }
        return count;
    }
};